include/deal.II/fe/fe_poly.h

```diff
@@ -281,13 +281,14 @@ class FE_Poly : public FiniteElement<dim, spacedim>
     // polynomial to put the values and derivatives of shape functions
     // to put there, depending on what the user requested
     std::vector<double> values(
-      update_flags & update_values ? this->n_dofs_per_cell() : 0);
+      (update_flags & update_values) != 0u ? this->n_dofs_per_cell() : 0);
```

Comment:
This particular change appears so many times that I'm wondering whether it would be useful to just rewrite this with a helper function, say
```suggestion
      update_flags_contain_bit(update_flags, update_values) ? this->n_dofs_per_cell() : 0);
```
or something similar. Not saying that you should do that, or that it should happen as part of this patch, just wondering. It might actually make the code easier to read.

Reply:
That would certainly go a ways towards treating the `UpdateFlags` enum as a true (set of) enumeration type, rather than as an integer type that is then bit-twiddled with.

Reply:
I will note that that will be as easy to replace after this patch as before, so let's just go ahead with this patch here.

Reply:
The problem is that we really get a mask back and not a flag (but they have the same type). A better way would likely be to use an enum class for the flags and a separate mask class and return a variable of mask class type when using `operator&` or `operator|`. That wouldn't be backward-compatible, though.

Reply:
I am late to this discussion, but I would add that I think that the recent change in terms of the update flags (and friends) here as well as in #12265 do not really help readability for me. One of the beauties we relied on in the past regarding the replacement of boolean arguments to enums with bit-wise values was the ability to write things like `if (update_flags & update_values)`, see for examples #10369.
However, the bitwise operator `&` is not the most obvious interface, especially for newcomers, so I would be strongly in favor of something more obvios. Not sure how exactly we would name it, as we have several more of these enum fields apart from `update_flags` so a uniform solution would be great, rather than `update_flags_contain_bit`. So something like `contain_bit` or `enabled(update_flags, update_values)` (my dream would be `update_flags.enable(update_values)` but that won't work with the plain enums).

Reply:
I agree with this.

Reply:
I personally find ``(update_flags & update_values)`` very clean and easy to understand and preferable over `` != 0u`` or a helper function. 

It is true that it is difficult to discover (as you can not look it up easily), but if you claim that ``(update_flags & update_values)`` is difficult to understand, you would need to get rid of ``update_values | update_gradients`` as well or not (which by the way, I do not suggest)?

Reply:
In my opinion, it makes sense to have both `update_flags & update_values)` and `contains(update_flags, update_values)` and I find the latter much more readable than the first if we are checking if certain flags are set (whereas we need `operator&` to modify flags).
Also, I find `operator|` much better readable than `operator&`, i.e. `if(contains(update_flags, update_values | update_gradients))` reads to me "if update_flags contain update_values or update_gradients".

I'm happy to put something like `contains(flags, flag)` together.

Reply:
Personally, I am in favor of reverting the changes regarding to the flags.

Reply:
The solution to what is discussed above is a "list-of-boolean" approach as described here https://github.com/dealii/dealii/pull/13290#issuecomment-1021311981